control-plane/api-gateway/controllers/finalizer.go

```diff
@@ -0,0 +1,41 @@
+package controllers
+
+import (
+	"context"
+
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// EnsureFinalizer ensures that the given object has the given finalizer.
+func EnsureFinalizer(ctx context.Context, client client.Client, object client.Object, finalizer string) (didUpdate bool, err error) {
```

Comment:
It's just there for clarity. When I return a boolean and the function isn't in the form of `IsXYZ`, I like to name it so that it's clear what that boolean represents. I don't use the naked return though because that is danger.

Reply:
oh yeah :-1:  on naked returns